packages/block-editor/src/components/block-patterns/index.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo, useCallback } from '@wordpress/element';
+import { parse } from '@wordpress/blocks';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { ENTER, SPACE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import BlockPreview from '../block-preview';
+
+function BlockPattern( { pattern, onClick } ) {
+	const { title, content } = pattern;
+	const blocks = useMemo( () => parse( content ), [ content ] );
+
+	return (
+		<div
```

Comment:
Unrelated to this PR, we probably should add more capabilities to the `Button` component:

```js
<Button
    as="div"
    className="block-editor-patterns__item"
    onClick={ () => onClick( blocks ) }
>
```

Behind the scenes, we would handle:
- `role`
- `onKeyDown`
- `tabIndex`

@diegohaz - is it how Reakit handles custom buttons? :)

Reply:
The Button has certain design choices in it, I don't see why I should use it here.

Reply:
> The Button has certain design choices in it, I don't see why I should use it here.

Such a button would have to live in `@wordpress/primitives`, I'm talking about accessibility aspects being sorted out. I forgot about visuals :)

Reply:
Off the top of my head, this is the list of things you should implement on non-native buttons:

- `role`
- `tabIndex`
- `onKeyDown` for <kbd>Enter</kbd> (trigger the button)
- `onKeyDown` for <kbd>Space</kbd> (activate the button, like a mouse down, `:active`)
- `onKeyUp` for `<kbd>Space</kbd>` (trigger the button just like <kbd>Enter</kbd>)
- `aria-disabled` (should add the attribute and call `.preventDefault()` and `.stopPropagation` on many events. Setting `pointer-events: none` may help with that, although it's not completely accurate).

Except for <kbd>Space</kbd>, that is handled like <kbd>Enter</kbd> (otherwise the code would be overcomplicated), Reakit's Button does all that. But it's much simpler to just use the native button. Why not use the native button here?

Reply:
If you don't plan to show this component disabled, I would say that the implementation is okay though.